### ISSUE TITLE
 [Feature Request] add 3des k1=k2,k1=k3,k2=k3, pw=16bytes

### ISSUE THREAD

**Issue by atarist** [hashcat/hashcat#2384]
is possible to change kernel to add this option for 14100

**Comment by philsmd** [comment]
In my optinion, even one of these variants would be quite challenging to get coded and tested correctly. Therefore, this is quite a huge request. I like one algorithm / variant issues much more, because when it's ready it could be closed.
I also do NOT think that we should/could do all of this in one kernel (hash type) because what if a user knows that it's the variant k1=k3 for sure... it would take 3 (or more) times as much with all those combinations done in one kernel. So my suggestion would be to use a separate hash type for each and every one of those variants.

If I look over here: https://en.wikipedia.org/wiki/Triple_DES#Keying_options , it seems that only these variants are more common (k1=k2=k3 , k1=k3 and of course k1 != k2 != k3). Therefore, I'm not so sure if the other one make much sense or should be prioritized, because there are only 3 common keying options (but I agree of course, every possible variant could be used by some software/algorithms nonetheless).

The main problem of course still remains, what is the use case for this and does this make the life of hashcat users easier ? The problem is that some people really have a hard time to see what is feasible and what is infeasible.

Brute-forcing 16 bytes (even using the reduced character set from hashcat: charsets/DES_full.charset) is and will remain infeasible. Just do the math: 128 ^ 16 = 5,192,296,858,534,827,628,530,496,329,220,096

This number of combination in **humongous**.

So we really should get some user feedback on how they are cleverly attacking (or are going to attack if this is implemented) their hashes. Is their strategy feasible ? Did they do the math ? How long would it take them to run the attack ? Does their strategy make sense ?

I know that the already available DES/3DES hash types suffer from pretty much exactly the same problem... but it was also meant to be run with a very limited set of unknown bytes.

Another possibility would be that the keys were generated by a known algorithm (not random) and therefore the 24 bytes, or with this variant the 16 bytes etc, are somehow weak or can be computed somehow. e.g. like a dictionary attack that was generated by some values originating from unix time () etc (an example of a non-random 24-byte key).

Nonetheless, we see this feature request quite a lot on the forum see:
- https://hashcat.net/forum/thread-9172.html
- https://hashcat.net/forum/thread-6582-post-35103.html#pid35103
- https://hashcat.net/forum/thread-6364.html

etc etc etc

You can try a quick feasibility test also by testing other hash types like this:
```
hashcat -a 3 -1 charsets/DES_full.charset --hex-charset 00000000000000000000000000000000 ?1?1?1?1?1?1?1?1
```

as you can see, if we increase the mask to ?1?1?1?1?1?1?1?1?1 (9 "random" bytes) it will already get infeasible and you will get this error message:
```
Integer overflow detected in keyspace of mask: ?1?1?1?1?1?1?1?1?1
```

Of course this is not a perfectly "valid" test, because we aren't using the correct hash type for this test etc.... but this way you will get a feeling how "exponential" the total keyspace (number of password candidates) will increase with increasing length of unknown bytes.

Would be great if we get some user feedback here and discuss if there is some very clever use case to use these variants and if the users tested if it would be feasible (and of course what time frame/resouces/power we are talking about here) .

Thx

**Comment by mfu-mcosys** [comment]
Hm, 1 year+ and still no 16byte / k1=k3, shit. I guess there is are huge demand on this. :)

**Comment by sll00** [comment]
Agreed, feasible or not it - it is the more feasible, and also fairly common version of 3DES. I'd offer a small bounty for implementation if it helped.
